lib/modules/manager/azure-pipelines/extract.ts

```diff
@@ -83,18 +93,64 @@ export function extractPackageFile(
   }
 
   // grab the repositories tags
-  for (const repository of pkg.resources.repositories) {
-    const dep = extractRepository(repository);
-    if (dep) {
-      deps.push(dep);
+  if (
+    !is.nullOrUndefined(pkg.resources) &&
+    is.nonEmptyArray(pkg.resources.repositories)
+  ) {
+    for (const repository of pkg.resources.repositories) {
+      const dep = extractRepository(repository);
+      if (dep) {
+        deps.push(dep);
+      }
     }
   }
 
   // grab the containers tags
-  for (const container of pkg.resources.containers) {
-    const dep = extractContainer(container);
-    if (dep) {
-      deps.push(dep);
+  if (
+    !is.nullOrUndefined(pkg.resources) &&
+    is.nonEmptyArray(pkg.resources.containers)
+  ) {
+    for (const container of pkg.resources.containers) {
+      const dep = extractContainer(container);
+      if (dep) {
+        deps.push(dep);
+      }
+    }
+  }
+
+  if (!is.nullOrUndefined(pkg.stages)) {
+    const tasks = pkg.stages
+      .flatMap((stage) =>
```

Comment:
please prefer `for...of` loops

Reply:
I think it gets quite deeply nested, but I've made the change.